### ISSUE TITLE
OpenDDSharp Interoperability

### ISSUE THREAD

**Issue by superware** [jmmorato/openddsharp#112]
Hello @jmmorato, great work!

I'm trying to setup a simple (all defaults) scenario using latest `OpenDDSSharp 0.8.21312.27` as publisher and `OpenDDS 3.2` as subscriber of the following:

```
module Devices {
	@topic
	struct Discovery {
		uint16 device_id;
	};
};
```

But when publishing, `on_requested_incompatible_qos` is being called in subscriber's `DataReaderListenerImpl` with `policy_id == 23` which seems to be `DATA_REPRESENTATION_QOS_POLICY_ID`.

Any ideas?

Thank you.

**Comment by superware** [comment]
Thanks, so what can be done with OpenDDS 3.2 in the meantime to be rtps_udp interoperable with 0.8.21312.27?

**Comment by jmmorato** [comment]
You can try with the new pre-release version following the new [Getting Started](https://jmmorato.github.io/openddsharp_documentation/articles/getting_started.html) article.

**Comment by superware** [comment]
Problem is I'm currently on .NET 4.7.2, plus having issues with building directly from Visual Studio 2022 instead of dotnet (which probably deserve a separate issue).

Anyway, that OpenDDS 3.16 note states "DataReaders will continue to be interoperable by default" so why OpenDDSharp (latest release) writer is incompatible with 3.2 reader? Even after explicitly setting the reader to CDR:

```
dr_qos.representation.value.length(1);
dr_qos.representation.value[0] = DDS::XCDR_DATA_REPRESENTATION;
```

on_requested_incompatible_qos is still being called. Is it possible to adjust 3.2 to work with 0.8.21312.27?

Thank you

**Comment by jmmorato** [comment]
> Anyway, that OpenDDS 3.16 note states "DataReaders will continue to be interoperable by default" 

I think what it means there is that, as only default encoding has changed for DataWriters, the DataReaders created with v3.16 and greater are still compatible with previous versions DataWriters. But this is not your case as I understood. You have the DataWriter (OpenDDSharp) with XCDR2 and the DataReader (OpenDDS 3.2) with XCDR  so they are not compatible. The only option is adapt your DataWriter because, as far as I know, XCDR2 was not implemented yet in OpenDDS 3.2.

What is your issue building with VS 2022?

**Comment by jmmorato** [comment]
Closed for inactivity.
Feel free to reopen it or create a new one with you issue.
